main.c

```diff
@@ -130,6 +130,16 @@ static int configure_network(const char *tapname,
         return -1;
     }
 
+    // set loopback device to UP
+    struct ifreq ifr_lo = {
+	    .ifr_name = "lo",
+	    .ifr_flags = IFF_UP | IFF_RUNNING
+    };
+    if (ioctl(sockfd, SIOCSIFFLAGS, &ifr_lo) < 0) {
+        perror("cannot set device up");
```

Comment:
If the interface is already up, does it return an error?

Reply:
nope!

Reply:
I tested it out, it does not!